### ISSUE TITLE
Option to not bump versions of already released crates

### ISSUE THREAD

**Issue by ascjones** [paritytech/subpub#4]
Recently we released new versions of `sp-core` and `sp-runtime` using subpub https://github.com/paritytech/substrate/pull/12599. A follow up was required to release a couple of higher level crates: https://github.com/paritytech/substrate/pull/12716.

However using `subpub` tried to bump all those already released crates again:

![image](https://user-images.githubusercontent.com/75586/202148282-31560ca3-73fe-41fd-89d8-d1efa405469a.png)

I did it manually in the end which was fairly simple, however it would be good to be able to have an option to not bump all those already released crates. Either explicitly exclude some crates by name e.g. `sp-core` or specify to *only* bump the crate to be released.

**Comment by dvdplm** [comment]
I'd even say not bumping and re-relasing already released crates should be the default? It seems like the most common behaviour.

/cc @joao-paulo-parity

**Comment by niklasad1** [comment]
> I'd even say not bumping and re-relasing already released crates should be the default? It seems like the most common behaviour.

That's implies that the tool has to check if "the crate's dependency graph" and compare each dep's version to previous release on crates.io if one exist otherwise do nothing. So that should be possible but require some work.

Alternatively, the tool can initially bump the versions of all crates which depends on the updated crates "not  only its dependencies". To elaborate if I want to bump `sp-core` with the tool sp-core is bumped and its dependencies but it may be x other crates that depend on `sp-core` with the new version on those crates are not bumped. Thus, it may end up in situation where a crate such as `sp-keyring` where the dependencies are already updated just not version of that create.

I'm still in favor what you are suggesting @dvdplm but `--exclude-crate foo` could also be an option.

**Comment by dvdplm** [comment]
> That's implies that the tool has to check "the crate's dependency graph" and compare each dep's version to previous release on crates.io if one exist otherwise do nothing. So that should be possible but requires some work.

Right, I somehow thought it already did this through the crates.io API. We are probably already using `crates_io_api` yeah? It looks like the [`Crate`](https://docs.rs/crates_io_api/latest/crates_io_api/struct.Crate.html#structfield.versions) stores the version as a `u64` so there'll be some conversion to do as well.

**Comment by jsdw** [comment]
FWIW I thought that it did check all dependents and only bumped versions when it detected some change in them (or some change in their dependencies which mean they need bumping too).
